From 2773cc95d0eae1a66a76b49e0b16f4a1d0bc1fa7 Mon Sep 17 00:00:00 2001
From: Davor Joja <davorjoja@logicbricks.com>
Date: Wed, 5 Nov 2014 16:51:02 +0100
Subject: [PATCH 166/456] gpu: drm: xylon: Removed pixel-format property

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Removed ABGR pixel-format property to avoid
non-standard change of crtc pixel format.

Signed-off-by: Davor Joja <davorjoja@logicbricks.com>
Tested-by: Christian Kohn <christian.kohn@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 871024032c77424477235a46cfc13991e1e7b345)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xylon/xylon_crtc.c          |   11 -----------
 drivers/gpu/drm/xylon/xylon_logicvc.c       |    7 -------
 drivers/gpu/drm/xylon/xylon_logicvc_layer.h |    4 +---
 drivers/gpu/drm/xylon/xylon_plane.c         |   17 -----------------
 drivers/gpu/drm/xylon/xylon_plane.h         |    1 -
 drivers/gpu/drm/xylon/xylon_property.c      |    6 ------
 drivers/gpu/drm/xylon/xylon_property.h      |    1 -
 7 files changed, 1 insertions(+), 46 deletions(-)

diff --git a/drivers/gpu/drm/xylon/xylon_crtc.c b/drivers/gpu/drm/xylon/xylon_crtc.c
index 6c041a6..a86bc98 100644
--- a/drivers/gpu/drm/xylon/xylon_crtc.c
+++ b/drivers/gpu/drm/xylon/xylon_crtc.c
@@ -43,7 +43,6 @@ struct xylon_drm_crtc_properties {
 	struct drm_property *control;
 	struct drm_property *color_transparency;
 	struct drm_property *interlace;
-	struct drm_property *pixel_format;
 	struct drm_property *transparency;
 	struct drm_property *transparent_color;
 	struct drm_property *position_x;
@@ -354,9 +353,6 @@ static int xylon_drm_crtc_set_property(struct drm_crtc *base_crtc,
 	} else if (property == props->interlace) {
 		op.id = XYLON_DRM_PLANE_OP_ID_INTERLACE;
 		op.param = (bool)val;
-	} else if (property == props->pixel_format) {
-		op.id = XYLON_DRM_PLANE_OP_ID_PIXEL_FORMAT;
-		op.param = (bool)val;
 	} else if (property == props->transparency) {
 		op.id = XYLON_DRM_PLANE_OP_ID_TRANSPARENCY;
 		op.param = val;
@@ -575,13 +571,6 @@ static int xylon_drm_crtc_create_properties(struct drm_crtc *base_crtc)
 					   "interlace",
 					   size))
 		return -EINVAL;
-	size = xylon_drm_property_size(property_pixel_format);
-	if (xylon_drm_property_create_list(dev, obj,
-					   &props->pixel_format,
-					   property_pixel_format,
-					   "pixel_format",
-					   size))
-		return -EINVAL;
 	if (transp_prop &&
 	    xylon_drm_property_create_range(dev, obj,
 					    &props->transparency,
diff --git a/drivers/gpu/drm/xylon/xylon_logicvc.c b/drivers/gpu/drm/xylon/xylon_logicvc.c
index 42d4b49..127a67f 100644
--- a/drivers/gpu/drm/xylon/xylon_logicvc.c
+++ b/drivers/gpu/drm/xylon/xylon_logicvc.c
@@ -125,7 +125,6 @@
 #define LOGICVC_LAYER_CTRL_ENABLE			(1 << 0)
 #define LOGICVC_LAYER_CTRL_COLOR_TRANSPARENCY_BIT	(1 << 1)
 #define LOGICVC_LAYER_CTRL_INTERLACE_BIT		(1 << 3)
-#define LOGICVC_LAYER_CTRL_PIXEL_FORMAT_BIT_ABGR	(1 << 4)
 
 /* logiCVC control registers initial values */
 #define LOGICVC_DTYPE_REG_INIT 0
@@ -474,12 +473,6 @@ void xylon_cvc_layer_ctrl(struct xylon_cvc *cvc, int id,
 	case LOGICVC_LAYER_INTERLACE_ENABLE:
 		regval &= ~LOGICVC_LAYER_CTRL_INTERLACE_BIT;
 		break;
-	case LOGICVC_LAYER_PIXEL_FORMAT_ABGR_DISABLE:
-		regval &= ~LOGICVC_LAYER_CTRL_PIXEL_FORMAT_BIT_ABGR;
-		break;
-	case LOGICVC_LAYER_PIXEL_FORMAT_ABGR_ENABLE:
-		regval |= LOGICVC_LAYER_CTRL_PIXEL_FORMAT_BIT_ABGR;
-		break;
 	default:
 		return;
 	}
diff --git a/drivers/gpu/drm/xylon/xylon_logicvc_layer.h b/drivers/gpu/drm/xylon/xylon_logicvc_layer.h
index b4cedb3..102ab35 100644
--- a/drivers/gpu/drm/xylon/xylon_logicvc_layer.h
+++ b/drivers/gpu/drm/xylon/xylon_logicvc_layer.h
@@ -21,9 +21,7 @@ enum xylon_cvc_layer_control {
 	LOGICVC_LAYER_COLOR_TRANSPARENCY_DISABLE,
 	LOGICVC_LAYER_COLOR_TRANSPARENCY_ENABLE,
 	LOGICVC_LAYER_INTERLACE_DISABLE,
-	LOGICVC_LAYER_INTERLACE_ENABLE,
-	LOGICVC_LAYER_PIXEL_FORMAT_ABGR_DISABLE,
-	LOGICVC_LAYER_PIXEL_FORMAT_ABGR_ENABLE
+	LOGICVC_LAYER_INTERLACE_ENABLE
 };
 
 struct xylon_cvc;
diff --git a/drivers/gpu/drm/xylon/xylon_plane.c b/drivers/gpu/drm/xylon/xylon_plane.c
index ba0791a..19f47be 100644
--- a/drivers/gpu/drm/xylon/xylon_plane.c
+++ b/drivers/gpu/drm/xylon/xylon_plane.c
@@ -35,7 +35,6 @@ struct xylon_drm_plane_properties {
 	struct drm_property *control;
 	struct drm_property *color_transparency;
 	struct drm_property *interlace;
-	struct drm_property *pixel_format;
 	struct drm_property *transparency;
 	struct drm_property *transparent_color;
 };
@@ -218,9 +217,6 @@ static int xylon_drm_plane_set_property(struct drm_plane *base_plane,
 	} else if (property == props->interlace) {
 		op.id = XYLON_DRM_PLANE_OP_ID_INTERLACE;
 		op.param = (bool)val;
-	} else if (property == props->pixel_format) {
-		op.id = XYLON_DRM_PLANE_OP_ID_PIXEL_FORMAT;
-		op.param = (bool)val;
 	} else if (property == props->transparency) {
 		op.id = XYLON_DRM_PLANE_OP_ID_TRANSPARENCY;
 		op.param = (u32)val;
@@ -275,13 +271,6 @@ static int xylon_drm_plane_create_properties(struct drm_plane *base_plane)
 					   "interlace",
 					   size))
 		return -EINVAL;
-	size = xylon_drm_property_size(property_pixel_format);
-	if (xylon_drm_property_create_list(dev, obj,
-					   &props->pixel_format,
-					   property_pixel_format,
-					   "pixel_format",
-					   size))
-		return -EINVAL;
 	if (!last_plane &&
 	    xylon_drm_property_create_range(dev, obj,
 					    &props->transparency,
@@ -461,12 +450,6 @@ int xylon_drm_plane_op(struct drm_plane *base_plane,
 		else
 			par = LOGICVC_LAYER_INTERLACE_DISABLE;
 		break;
-	case XYLON_DRM_PLANE_OP_ID_PIXEL_FORMAT:
-		if (op->param)
-			par = LOGICVC_LAYER_PIXEL_FORMAT_ABGR_ENABLE;
-		else
-			par = LOGICVC_LAYER_PIXEL_FORMAT_ABGR_DISABLE;
-		break;
 	}
 
 	xylon_cvc_layer_ctrl(cvc, id, par);
diff --git a/drivers/gpu/drm/xylon/xylon_plane.h b/drivers/gpu/drm/xylon/xylon_plane.h
index 40ad52b..c61c0a4 100644
--- a/drivers/gpu/drm/xylon/xylon_plane.h
+++ b/drivers/gpu/drm/xylon/xylon_plane.h
@@ -24,7 +24,6 @@ enum xylon_drm_plane_op_id {
 	XYLON_DRM_PLANE_OP_ID_BACKGROUND_COLOR,
 	XYLON_DRM_PLANE_OP_ID_COLOR_TRANSPARENCY,
 	XYLON_DRM_PLANE_OP_ID_INTERLACE,
-	XYLON_DRM_PLANE_OP_ID_PIXEL_FORMAT,
 	XYLON_DRM_PLANE_OP_ID_TRANSPARENCY,
 	XYLON_DRM_PLANE_OP_ID_TRANSPARENT_COLOR
 };
diff --git a/drivers/gpu/drm/xylon/xylon_property.c b/drivers/gpu/drm/xylon/xylon_property.c
index 3321ebe..6cd71a8 100644
--- a/drivers/gpu/drm/xylon/xylon_property.c
+++ b/drivers/gpu/drm/xylon/xylon_property.c
@@ -54,12 +54,6 @@ const struct drm_prop_enum_list property_interlace[] = {
 	{/* end */}
 };
 
-const struct drm_prop_enum_list property_pixel_format[] = {
-	{ 0, "Plane ABGR Format Disable" },
-	{ 1, "Plane ABGR Format Enable" },
-	{/* end */}
-};
-
 int xylon_drm_property_size(const struct drm_prop_enum_list *list)
 {
 	int i = 0;
diff --git a/drivers/gpu/drm/xylon/xylon_property.h b/drivers/gpu/drm/xylon/xylon_property.h
index e814f6e..a98248a 100644
--- a/drivers/gpu/drm/xylon/xylon_property.h
+++ b/drivers/gpu/drm/xylon/xylon_property.h
@@ -28,7 +28,6 @@ extern const struct drm_prop_enum_list property_pixel_data_trigger[];
 extern const struct drm_prop_enum_list property_control[];
 extern const struct drm_prop_enum_list property_color_transparency[];
 extern const struct drm_prop_enum_list property_interlace[];
-extern const struct drm_prop_enum_list property_pixel_format[];
 
 int xylon_drm_property_size(const struct drm_prop_enum_list *list);
 
-- 
1.7.5.4

